service/controller/resource/instance/create_cordon_old_workers.go

```diff
@@ -165,3 +165,59 @@ func sortNodesByTenantVMState(nodes []corev1.Node, instances []compute.VirtualMa
 
 	return
 }
+
+func (r *Resource) getK8sWorkerNodeForInstance(ctx context.Context, customObject providerv1alpha1.AzureConfig, instance compute.VirtualMachineScaleSetVM) (*corev1.Node, error) {
```

Comment:
Can I propose to move these functions to e.g. `worker.go`? Now they are in one of the state transition files and then used in the others. When looking for code it's confusing even though editor / IDE helps finding it. :slightly_smiling_face:

Reply:
What about having _worker node_ <-> _VMSS instance_ mapping saved (in status?) so we can look it up when we need it?
(maybe too much/unnecessary for this fix)